### DISCUSSION TITLE
[Next.JS Integration] making webpack-nextjs sample project work

### DISCUSSION THREAD

**Discussion by derais** [CoderLine/alphaTab#1550] [General]
**[Link to the "webpack-nextjs" sample project in question](https://github.com/CoderLine/alphaTabSamplesWeb/tree/main/src/webpack-nextjs)** 

**Related update:  #1539** 

Recently found out the update that fixes the next.js implementation and the sample project along with it and I'm really happy it works! Doesn't work straight out the box, though. I originally made this discussion trying to find out how to fix it but I figured it out, so just here to share.

The fix involves simply moving the font and soundfont to the public folder. I'm not sure if I was missing anything but this fixed all the problems with the project.

It seems there might a missing file `src/alphaTab.ts` in the sample project as mentioned in the [README.md](https://github.com/CoderLine/alphaTabSamplesWeb/tree/main/src/webpack-nextjs), or at least I can't find them. I can't find it in [alphaTab/src/webpack](https://github.com/CoderLine/alphaTab/tree/639400bb1ba4d60e4b0457c31977619e90bec801/src/webpack) either. Wonder what it does, but the project seems to work fine.

### Initial Setup:
```
1. git clone https://github.com/CoderLine/alphaTabSamplesWeb.git
2. /src/webpack-nextjs
3. pnpm install
4. pnpm dev
```
### Initial Output:
![image](https://github.com/CoderLine/alphaTab/assets/40361361/2232cea4-228e-4d6b-a1b1-f6b5c10cb35f)

### Fixed Output
![image](https://github.com/CoderLine/alphaTab/assets/40361361/0f4f5dd2-f1b6-46f3-8ebf-3e0117a646d3)

**Comment by Danielku15** [comment]
Seems there is still a slight problem with the copying of assets (web and sound fonts). 

The WebPack plugin copies the files (to the output directory) and registers them as assets into WebPack. Normally this ensures they are served by the dev server.

In case of next.js `npm run dev` this seem to fail:

1. The output path is set to `node_modules/.next/` and you will see the font and soundfont directories there. 
2. But when trying to open the paths, WebPack doesn't serve the assets but a 404 page is shown. 

This is a known problem in next.js: 

https://github.com/vercel/next.js/discussions/32836
https://github.com/vercel/next.js/issues/45478

I updated the sample at https://github.com/CoderLine/alphaTabSamplesWeb/tree/main/src/webpack-nextjs to work.
